ansible_rulebook/vault.py

```diff
@@ -0,0 +1,106 @@
+#  Copyright 2024 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import getpass
+import shutil
+import tempfile
+
+import pexpect
+
+from ansible_rulebook.exception import (
+    AnsibleVaultNotFound,
+    VaultDecryptException,
+)
+
+VAULT_HEADER = "$ANSIBLE_VAULT"
+b_VAULT_HEADER = b"$ANSIBLE_VAULT"
+
+
+class Vault:
+    """Vault class allows for decryption."""
+
+    def __init__(
+        self,
+        passwords: list[dict] = None,
+        password_file: str = None,
+        vault_ids: list[str] = None,
+        ask_pass: bool = False,
+    ):
+        cli_args = ""
+        if ask_pass:
+            self.secret = getpass.getpass(prompt="Vault password: ")
```

Comment:
This is one time thing. We ask a password only if the cli has included --ask-vault-pass flag.

Reply:
Well, first at all, I don't like the idea of such a flag in ansible-rulebook because unlike ansible, ansible-rulebook is not intended to be an interactive process. But if we are going to implement it because ansible have it,  we should replicate the behavior of ansible. Ansible asks for every vault id provided. Or we should include a check to accept only the interactive password only if not id's are provided.

Reply:
Each --vault-id must be followed by a password file. It does not ask the user to type a password in the console. Can you give an example how ansible asks for every vault-id?

Reply:
ok, I was my misunderstanding. It supports only a single password.